.github/actions/artifact_download/action.yml

```diff
@@ -0,0 +1,55 @@
+name: "Download Artifacts"
+description: "Download build Result Binaries"
+inputs:
+  profile:
+    description: "debug/release"
+    required: true
+  sha:
+    description: "git sha"
+    required: true
+  target:
+    description: ""
+    required: true
+  source:
+    description: "github/s3"
+    required: true
+  category:
+    description: "default/hive"
+    required: false
+    default: default
+  path:
+    description: "default to ./target/{inputs.profile}/"
+    required: false
+    default: ""
+runs:
+  using: "composite"
+  steps:
+    - name: Get Download Path
+      id: path
+      shell: bash
+      run: |
+        if [[ -z "${{ inputs.path }}"]]; then
+          path="./target/${{ inputs.profile }}"
+        else
+          path="${{ inputs.path }}"
+        fi
+        echo "::set-output name=path::${path}"
+
+    - uses: actions/download-artifact@v2
+      if: inputs.source == 'github'
+      with:
+        name: ${{ inputs.profile }}-${{ inputs.sha }}-${{ inputs.target }}-${{ inputs.category }}
+        path: ${{ steps.path.outputs.path }}/
+
+    - name: Download artifact from s3
+      if: inputs.source == 's3'
+      shell: bash
+      run: |
+        aws s3 cp \
+          s3://databend-ci/${{ inputs.profile }}/${{ inputs.sha }}/${{ inputs.target }}/${{ inputs.category }}/ \
```

Comment:
Have we set the lifecycle for those artifacts?

Reply:
> Have we set the lifecycle for those artifacts?

Yes, the lifecycle is 7 days currently.